pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -190,31 +190,28 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	updated := false
 	for _, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
 		for _, snapshot := range policyVolumeSnapshot {
-			// Get the updated status if we see it as not completed
-			if !s.isVolumeSnapshotComplete(snapshot.Status) {
-				pendingVolumeSnapshotStatus, err := getVolumeSnapshotStatus(snapshot.Name, snapshotSchedule.Namespace)
-				if err != nil {
-					return err
-				}
+			pendingVolumeSnapshotStatus, err := getVolumeSnapshotStatus(snapshot.Name, snapshotSchedule.Namespace)
+			if err != nil {
+				return err
+			}
 
-				// Check again and update the status if it is completed
-				snapshot.Status = pendingVolumeSnapshotStatus
-				if s.isVolumeSnapshotComplete(snapshot.Status) {
-					snapshot.FinishTimestamp = meta.NewTime(schedule.GetCurrentTime())
-					if pendingVolumeSnapshotStatus == snapv1.VolumeSnapshotConditionReady {
-						s.recorder.Event(snapshotSchedule,
-							v1.EventTypeNormal,
-							string(snapv1.VolumeSnapshotConditionReady),
-							fmt.Sprintf("Scheduled snapshot (%v) completed successfully", snapshot.Name))
-					} else {
-						s.recorder.Event(snapshotSchedule,
-							v1.EventTypeWarning,
-							string(snapv1.VolumeSnapshotConditionError),
-							fmt.Sprintf("Scheduled snapshot (%v) failed", snapshot.Name))
-					}
+			// Check again and update the status if it is completed
+			snapshot.Status = pendingVolumeSnapshotStatus
+			if s.isVolumeSnapshotComplete(snapshot.Status) {
+				snapshot.FinishTimestamp = meta.NewTime(schedule.GetCurrentTime())
+				if pendingVolumeSnapshotStatus == snapv1.VolumeSnapshotConditionReady {
+					s.recorder.Event(snapshotSchedule,
+						v1.EventTypeNormal,
+						string(snapv1.VolumeSnapshotConditionReady),
+						fmt.Sprintf("Scheduled snapshot (%v) completed successfully", snapshot.Name))
+				} else {
+					s.recorder.Event(snapshotSchedule,
+						v1.EventTypeWarning,
+						string(snapv1.VolumeSnapshotConditionError),
+						fmt.Sprintf("Scheduled snapshot (%v) failed", snapshot.Name))
 				}
-				updated = true
 			}
+			updated = true
```

Comment:
Wouldn't this end up updating the VolumeSnapshotSchedule endlessly? The `updated` boolean should be set only if snapshotStatus != pendingVolumeSnapshotStatus before line 199

Reply:
yes,  we shouldn't retry for `ready` snapshot. I think we can omit update of `volumesnapshotschedule` is latest snapshotstatus is same as entry in `volumesnapshotschedule` object